snuba/datasets/discover.py

```diff
@@ -215,16 +250,18 @@ def __init__(
                     ColumnToMapping(None, "release", None, "tags", "sentry:release"),
                     ColumnToMapping(None, "dist", None, "tags", "sentry:dist"),
                     ColumnToMapping(None, "user", None, "tags", "sentry:user"),
-                    DefaultNoneColumnMapper(self.__abstract_transactions_columns),
+                    DefaultColumnMapper(self.__abstract_transactions_columns),
                 ]
             )
         )
 
         self.__transaction_translator = transaction_translator.concat(
             TranslationMappers(
                 columns=[
-                    ColumnToLiteral(None, "group_id", 0),
-                    DefaultNoneColumnMapper(self.__abstract_events_columns),
+                    DefaultColumnMapper(
+                        ColumnSet([("group_id", Nullable(UInt(64)))]), 0
+                    ),
```

Comment:
What is the difference with respect to what we were having before? It seems we are still turning the group_id column into 0.

Reply:
We're mapping the column to 0 not just in the selected_columns statement, so that the example you gave earlier with group ID in a condition `type=transaction AND group_id=123` also works

Reply:
Yes, but the previous `ColumnToLiteral(None, "group_id", 0),` should already have done this. I am not sure what changes with respect to the previous implementation.

Reply:
I don't think it does - if so why did that query previously 500 like you mentioned in https://github.com/getsentry/snuba/pull/1289#discussion_r483310746

Reply:
I think I was wrong on that comment. the query would have gone to the transactions table and group_id would have been transformed into 0.
The point kind of remains, we are making a functional change in the result, which is fine, but I would collect data silently without changing the selected table before making the dataset behave differently.

Reply:
That's fine. Just to be clear this was always the original behaviour though that has been running most of the time with Discover. It was only changed very recently with the AST roll out and this PR only reverts that change. No other result should be different yet.

Reply:
The only part of this change that will potentially affect users is that it reverts the table selection behavior back to what we had prior to https://github.com/getsentry/snuba/commit/ea1a2776d337ab8b6ecf1424cd830e412fbe158b

Reply:
Fair point. If Evan is ok with this change that's good to me. Though I think you can revert the change to the translator of the group_id column.

Reply:
reverted